internal/config/concierge/config.go

```diff
@@ -186,6 +191,22 @@ func validateNames(names *NamesConfigSpec) error {
 	return nil
 }
 
+func validateKubeCertAgent(agentConfig *KubeCertAgentSpec) error {
+	if len(agentConfig.PriorityClassName) == 0 {
+		// Optional, so empty is valid.
+		return nil
+	}
+
+	// See https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass
+	// for PriorityClassName rules.
+	errStrings := validation.IsDNS1123Subdomain(agentConfig.PriorityClassName)
```

Comment:
Could you help me understand why we should pre-validate this? Why not allow the deployment to proceed with an invalid `PriorityClassName` and have the deployment status tell us what the issue was? We're also not checking to make sure that the name isn't prefixed with `system-`.